packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts

```diff
@@ -212,6 +171,19 @@ function newView(
     version,
   })
 
+  let scheduleStopInitialViewTimingsTracking: () => void
+  if (loadingType === ViewLoadingType.INITIAL_LOAD) {
+    ;({ scheduleStop: scheduleStopInitialViewTimingsTracking } = trackInitialViewTimings(lifeCycle, (newTimings) => {
+      timings = newTimings
+      if (newTimings.loadEvent !== undefined) {
+        setLoadEvent(newTimings.loadEvent)
+      }
+      scheduleViewUpdate()
+    }))
+  } else {
+    scheduleStopInitialViewTimingsTracking = noop
+  }
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏there is a lot going there, it could be interesting to extract a function for the timings update. 
After reducing the complexity, using a ternary there could also remove a bit of boilerplate, something like:

```ts
const { scheduleStop: scheduleStopInitialViewTimingsTracking } = loadingType === ViewLoadingType.INITIAL_LOAD
    ? trackInitialViewTimings(lifeCycle, updateViewTimings)
    : { scheduleStop: noop }

...

function updateViewTimings(...) {
  ...
}
```

Reply:
Even better!